woocommerce/woocommerce-ios
# Store creation: add support CTA to applicable screens

Part of #8538 


## Description


From p1670931902787739/1670929798.889469-slack-C045CUK1Y3U, we want to make it easier for users to contact support during the store creation flow. We're adding a ❓ icon to the navigation bar trailing edge to screens that don't have a trailing bar button - store name, country selector, domain selector, and store summary.

I created a SwiftUI view `SupportButton` to show as a `ToolbarItem` at the trailing edge of the navigation bar in the 4 screens with some refactoring on the related code. A new image `helpOutline` is added following the [Android implementation](https://github.com/woocommerce/woocommerce-android/pull/8054).

Please note that I'm clarifying two points on this UX change p1672818334782819-slack-C045CUK1Y3U, there might be some changes after the decision is made.

## Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the store name form should be shown with a support CTA on the top right of the navigation bar
- Tap on the icon to submit a support ticket --> the Zendesk ticket should have the tag `origin:store-creation` (make sure to note that the ticket is only for testing since we don't have permission to close them in Zendesk)
- Go back and enter a store name, then tap `Continue` --> the category question should be shown without a support CTA
- Tap `Continue` to skip --> the selling status question should be shown without a support CTA
- Tap `Continue` to skip --> the country question should be shown with a support CTA on the top right of the navigation bar
- Tap on the icon to submit a support ticket --> the Zendesk ticket should have the tag `origin:store-creation`
- Go back and select an option, then tap `Continue` --> the domain selector should be shown with a support CTA on the top right of the navigation bar
- Tap on the icon to submit a support ticket --> the Zendesk ticket should have the tag `origin:store-creation`
- Go back and select a domain, then tap `Continue` --> the summary screen should be shown with a support CTA on the top right of the navigation bar
- Tap on the icon to submit a support ticket --> the Zendesk ticket should have the tag `origin:store-creation`

## Screenshots


store name | country selector | domain selector | store summary
-- | -- | -- | --
![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-04 at 15 33 54](https://user-images.githubusercontent.com/1945542/210522901-698b5114-2bda-4621-afe0-42e02515f9a4.png) | ![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-04 at 15 34 00](https://user-images.githubusercontent.com/1945542/210522919-caa73db6-7fc3-46eb-8854-623d60ec3b48.png) | ![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-04 at 15 34 05](https://user-images.githubusercontent.com/1945542/210522923-0b5bf141-aa05-4484-8c12-6aed8ab95cbf.png) | ![Simulator Screen Shot - iPhone 14 Pro Max - 2023-01-04 at 15 34 14](https://user-images.githubusercontent.com/1945542/210522926-64646dc1-8244-4be8-989c-70f9b2c1b22f.png)


---
